Assets/Scripts/Audio/AudioController.cs

```diff
@@ -0,0 +1,106 @@
+﻿using UnityEngine;
+using System.Collections;
+using System.Collections.Generic;
+
+
+public class AudioController : MonoBehaviour {
+
+    private float maxVolume;
+	private float fadeTime;
+
+	public bool hasPlayer;
+
+    private BoxCollider2D audioRangeCollider;
+    private AudioSource audioSource;
+    public List<GameObject> dolls = new List<GameObject>();
```

Comment:
I also don't think we need the dolls list anymore.

If the collider rechecks for dolls that change tags. Then all you need to do is trigger FadeIn when a doll is detected by OnTriggerEnter. There is no need to keep the dolls in a list.

Why I had the list before was that so we could keep track of all dolls in order that if one changes from NonPlayer-> Player without exiting the collider, it will find this Player doll on the next update.